src/remoteConnector.ts

```diff
@@ -761,6 +768,15 @@ export default class RemoteConnector extends Disposable {
 		}
 	}
 
+	private async startHeartBeat(connectionInfo: SSHConnectionParams) {
+		if (this.heartbeatManager?.instanceId !== connectionInfo.instanceId) {
```

Comment:
I wonder why do we need thid guard? It should be called only once per window?

Reply:
yes but just in case :slightly_smiling_face: